### PR TITLE
Skip processing

### PR DESCRIPTION
**Pull request by martonilles**
closes: #262

### CODE REVIEW

File: unblob/processing.py [hunk 1]
```diff
@@ -146,6 +149,18 @@ def __init__(
...
+            [detect.name.startswith(pattern) for pattern in self.config.skip_magic]
+        )
+        if skip:
+            logger.debug(
+                "Skip file from processing based on file-magic",
+                magic=detect.name,
+                path=self.task.path,
+            )
+            return
+
```
Comment (vlaci) [inline unblob/processing.py:163]: For me it'd make sense to extract this into a `_should_skip` method inside `Processor` and call from the `if..elif..` chain in `process_task`. That way all the logic determining if we want to process a file would be in the same place.

File: unblob/processing.py [hunk 2]
```diff
@@ -146,6 +149,18 @@ def __init__(
     def process(self):
         logger.debug("Processing file", path=self.task.path, size=self.size)
 
+        detect = magic.detect_from_filename(self.task.path)
+        skip = any(
+            [detect.name.startswith(pattern) for pattern in self.config.skip_magic]
+        )
```
Comment (vlaci) [inline unblob/processing.py:155]: You can get rid of the list comprehension and make the `any` call short-circuiting that way:
```suggestion
        skip = any(
            detect.name.startswith(pattern) for pattern in self.config.skip_magic
        )
```

```
>>> def is_true():
...     print("It is True!")
...     return True
... 
>>> any([is_true() for _ in range(3)])
It is True!
It is True!
It is True!
True
>>> any(is_true() for _ in range(3))
It is True!
True
>>> 
```

**Review by vlaci — CHANGES_REQUESTED** [review]

**Comment by vlaci** [comment]
Very nice!

File: unblob/processing.py (hunk 1 above)
Comment (martonilles) [reply unblob/processing.py]: done

File: unblob/processing.py (hunk 2 above)
Comment (martonilles) [reply unblob/processing.py]: good point :)

**Review by vlaci — APPROVED** [review]

**Pull request merged**